docs/_articles/en/v3.x/salesforce-graph-engine/working-with-sfge.md

```diff
@@ -80,6 +80,36 @@ Example usage:
 public class MyClass {
 ```
 
+## "OutOfMemory: Java heap space" Error
+
+The number of paths SFGE needs to create increases exponentially with every conditional or method invocation in your code. As the number of paths explode, SFGE's probability of encountering an OutOfMemory error increases. In addition, a variety of factors such as your OS type, java setup, and other processes running in your machine can influence the heap space assigned by JVM, which then impacts SFGE's execution.
+
+Even if SFGE's execution was interrupted, it returns results from the portion of source code it has analyzed so far.
+
+While we are working on optimizing SFGE's heap consumption, here are some tips to avoid an OutOfMemory error:
+1. Increase max heap size assigned to SFGE's execution. You can do this by providing an updated `-Xmx` value to either the `sfgejvmargs` parameter with `scanner:run:dfa` command, or to the environment variable, `SFGE_JVM_ARGS`.
+
+	Before you do this, make note of your JVM's default heap size:
+	- Windows: `javaw -XX:+PrintFlagsFinal | find "MaxHeapSize"`
+	- Linux: `jstat -gccapacity <insert-sfge-pid-here>`
```

Comment:
How does one get the sfge pid? That feels like it should be included.

Reply:
I feel like we are already providing too much information that users can easily google and find out themselves. These commands are only meant to be guidelines.
Also, I haven't tested the Windows and Linux commands. Without going through the process ourselves, I don't feel comfortable adding more information that's outside the realm of our product.